src/opf/ed.jl

```diff
@@ -71,52 +67,35 @@ function build_opf(::Type{OPF}, network::Dict{String,Any}, optimizer;
 
     # Generation bounds (zero if generator is off)
     set_lower_bound.(pg, gen_status .* pgmin)
-    set_upper_bound.(pg, gen_status .* pgmax)
+    # ⚠️ we do not set upper bounds on pg because they are redundant
+    #       with the constraint pg + r <= pgmax
 
-    if OPF == EconomicDispatchWithReserves
-        # Reserve bounds (zero if generator is off)
-        set_lower_bound.(r, gen_status .* rmin)
-        set_upper_bound.(r, gen_status .* rmax)
-    end
+    # Reserve bounds (zero if generator is off)
+    set_lower_bound.(r, gen_status .* rmin)
+    set_upper_bound.(r, gen_status .* rmax)
 
     # Flow and flow slack bounds
-    for e in 1:E if !branch_status[e]
-            set_upper_bound(δf[e], 0.0)
-        end
-    end
```

Comment:
I removed this because
* we throw an error earlier if there are inactive branches
* we shouldn't need to set an upper bound anyway, penalty variables will be zero at the optimum